sw/airborne/modules/optical_flow/mateksys_3901_l0x.c

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright (C) 2020 Paparazzi Team
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ */
+
+/** @file modules/optical_flow/mateksys_3901_l0x.h
+ *  @brief Driver for the mateksys_3901_l0x sensor via MSP protocol output
+ *
+ */
+
+#include "mateksys_3901_l0x.h"
+#include "mcu_periph/uart.h"
+#include "subsystems/abi.h"
+
+// State interface for rotation compensation
+#include "state.h"
+
+// Messages
+#include "pprzlink/messages.h"
+#include "subsystems/datalink/downlink.h"
+
+
+// Define configuration parameters
+#ifndef MATEKSYS_3901_L0X_MOTION_THRES
+#define MATEKSYS_3901_L0X_MOTION_THRES
+#endif
+
+#ifndef MATEKSYS_3901_L0X_DISTANCE_THRES
+#define MATEKSYS_3901_L0X_DISTANCE_THRES
+#endif
+
+#ifndef USE_MATEKSYS_3901_L0X_AGL
+#define USE_MATEKSYS_3901_L0X_AGL
+#endif
+
+#ifndef USE_MATEKSYS_3901_L0X_OPTICAL_FLOW
+#define USE_MATEKSYS_3901_L0X_OPTICAL_FLOW
+#endif
+
+#ifndef MATEKSYS_3901_L0X_COMPENSATE_ROTATION
+#define MATEKSYS_3901_L0X_COMPENSATE_ROTATION
+#endif
+
+struct Mateksys3901l0X mateksys3901l0x = {
+  .parse_status = MATEKSYS_3901_L0X_INITIALIZE
+};
+
+static void mateksys3901l0x_parse(uint8_t byte);
+
+#if PERIODIC_TELEMETRY
+#include "subsystems/datalink/telemetry.h"
+
+/**
+ * Downlink message flow and lidar (included velocity estimation, not yet tested)
+ */
+static void mateksys3901l0x_send_matek(struct transport_tx *trans, struct link_device *dev)
+{
+  pprz_msg_send_MATEKSYS_FLOW_LIDAR(trans, dev, AC_ID,
+                                    &mateksys3901l0x.sensor_id,
+                                    &mateksys3901l0x.motion_quality,
+                                    &mateksys3901l0x.motionX_clean,
+                                    &mateksys3901l0x.motionY_clean,
+                                    &mateksys3901l0x.distancemm_quality,
+                                    &mateksys3901l0x.distance_clean,
+                                    &mateksys3901l0x.velocityX,
+                                    &mateksys3901l0x.velocityY);
```

Comment:
Actually, the only missing element now is this message from pprzlink. As mentioned in https://github.com/paparazzi/pprzlink/pull/137, I'd like to merge some optical flow message and avoid a dedicated one per sensor if possible. Once the pprzlink PR is accpeted and this PR updated accordingly (message and reference to pprzlink submodule), it is good to merge.

Reply:
done. About the reference to the submodule, the new ppzlink branch is on a fork, so maybe the PR of ppzlink should be merged to master first?

Reply:
yes, once pprzlink message is in master, you can update submodule ref here